src/libaktualizr/uptane/managedsecondary.cc

```diff
@@ -104,6 +105,14 @@ bool ManagedSecondary::putRoot(const std::string &root, const bool director) {
 }
 
 bool ManagedSecondary::sendFirmware(const std::string &data) {
+  if (install_future.wait_for(std::chrono::seconds(0)) == std::future_status::ready) {
+    install_future = std::async(std::launch::async, &ManagedSecondary::sendFirmware_, this, std::ref(data));
+    return true;
+  }
+  return false;
+}
```

Comment:
Nice. Just to make sure I understand: this checks if the installation thread is available. If so, it tries to install. If not, it aborts and returns false. Is there any sort of timeout mechanism for the installation? (We don't necessarily need that now, I'm just checking.)

Reply:
Yes, it just checks if thread is running or not. No we don't  have any timeout mechanisms now.